### PR TITLE
413 keep lastupdateddate col (staging fix)

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

Feedback from the testers (see Jira ticket 413) identified the following:

1) The business area wants column `lastupdateddate` to be retained, this is now not dropped from the contributors dataframe. 

This takes place in `staging/spp_snapshot_processing.py` and the corresponding test has been updated. 

NOTE: by default, the datatype for lastupdateddate, datetime64, writes to Excel in a strange, unreadable format. For this reason, I've added a `date_format` parameter to `to_csv` in the `write_local_csv` and `write_hdfs_csv` functions, please have a look.

2) The columns `frozensicoutdated`, `rusicoutdated`, `frozensic`, `rusic` were missing leading zeros in Excel. 
We considered ensuring these were strings in the pipeline, but this could cause conflicts with mapping to SIC codes where these are read in from csv (in which case they might not contain the leading zero). So we decided to keep these values as integers.
So no change here.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #413

#### Code

- [x] **Code runs** The code runs on my machine and/or CDSW
- [x] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [x] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [x] **Clean Code**
    - [x] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [x] Code passess flake8 linting check
    - [x] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [x] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [x] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/utils/local_file_mods.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def write_local_csv(filepath: str, data: pd.DataFrame):
     # Open the file in write mode
     with open(filepath, "w", newline="\n", encoding="utf-8") as file:
         # Write dataframe to the file
-        data.to_csv(file, index=False)
+        data.to_csv(file, date_format="%Y-%m-%d %H:%M:%S.%f+00", index=False)
```
Comment (smithv1) [inline src/utils/local_file_mods.py:47]: Tested by add;
         data["timestamp"] = pd.to_datetime(data["timestamp"], format="%d/%m/%Y-%H:%M:%S")
         data["timestamp"] = data["timestamp"].dt.strftime("%m/%d/%Y-%H:%M:%S")

And got an empty data error - not sure if there's a better way to test this

**Review by smithv1 — COMMENTED** [review]
Code looks good, test pass and runs fine. Would like to check if there's another way to test the code: "date_format="%Y-%m-%d %H:%M:%S.%f+00", index=False)"

File: src/utils/local_file_mods.py (hunk 1 above)
Comment (AnneONS) [reply src/utils/local_file_mods.py:47]: Many thanks for looking at this. I chose this date_format by looking at the `Description = "Datetime format = %Y-%m-%d %H:%M:%S.%f+00"` in  the `contributors_schema.toml`. I tested the output looked good in the full_responses output, and noted that it was the same format as in the original (old) version of the contributors csv given by the business area.

Comment (smithv1) [reply src/utils/local_file_mods.py:47]: Ran with and without and got expected output so good to merge this one now

**Review by smithv1 — APPROVED** [review]

**Pull request merged**